x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/TDigestArrayBlock.java

```diff
@@ -246,6 +246,41 @@ void copyInto(
         valueCountsBuilder.copyFrom(valueCounts, beginInclusive, endExclusive);
     }
 
+
+    @Override
+    public DoubleBlock buildHistogramComponentBlock(Component component) {
+        // as soon as we support multi-values, we need to implement this differently,
+        // as the sub-blocks will be flattened and the position count won't match anymore
+        // we'll likely have to return a "view" on the sub-blocks that implements the multi-value logic
+        assert doesHaveMultivaluedFields() == false;
+        return switch (component) {
+            case MIN -> {
+                minima.incRef();
+                yield minima;
+            }
+            case MAX -> {
+                maxima.incRef();
+                yield maxima;
+            }
+            case SUM -> {
+                sums.incRef();
+                yield sums;
+            }
+            case COUNT -> {
+                try(var doubleBuilder = blockFactory().newDoubleBlockBuilder(valueCounts.getPositionCount())) {
```

Comment:
In order to not have to deal with a special case of `COUNT` being long instead of double for `TDigest` we convert to a `DoubleBlock` here.
We'll likely switch the internal representation later to `DoubleBlock` anyway in order to deal with extrapolation in `increase()` and for `rate()`.